ground/gcs/src/plugins/uavtalk/telemetry.cpp

```diff
@@ -398,7 +398,12 @@ void Telemetry::transactionTimeout(ObjectTransactionInfo *transInfo)
     }
 }
 
-QByteArray *Telemetry::downloadFile(quint32 fileId, quint32 maxSize)
+/* This is synchronous, so we use a primitive callback mechanism
+ * instead of signal/slot.  Can have a future async variant if
+ * necessary
```

Comment:
Probably for some things.

I really like intuitive code describing the sequence things will happen in instead of ... signals cascading to each other willy nilly and efforts to jam a crude state machine onto them.  So even if this becomes async, I think we'd want one level up to be synchronous/fairly imperative.

But conversely, we can use a lot of stack if there is any real concurrency between things using this model.